docs/datagen/client/localization.md

```diff
@@ -0,0 +1,27 @@
+Language Generation
+===================
+
+[Language files][lang] can be generated for a mod by subclassing `LanguageProvider` and implementing `#addTranslations`. Each `LanguageProvider` subclass created represents a separate [locale] (`en_us` represents American English, `es_es` represents Spanish, etc.). After implementation, the provider must be [added][datagen] to the `DataGenerator`.
+
+`LanguageProvider`
+------------------
+
+Each language provider is simple a map of strings where each translation key is mapped to a localized name. A translation key mapping can be added using `#add`. Additionally, there are methods which use the translation key of a `Block`, `Item`, `ItemStack`, `Enchantment`, `MobEffect`, and `EntityType`.
+
+```java
+// In LanguageProvider#addTranslations
+this.addBlock(EXAMPLE_BLOCK, "Example Block");
+this.add("object.examplemod.example_object", "Example Object");
+```
+
+!!! tip
+    Localized names which contain alphanumeric values not in American English can be supplied as is. The provider automatically translates the characters into their unicode equivalents to be read by the game.
+
+    ```java
+    // Encdoded as 'Art\u00EDculo de ejemplo
+    this.addItem(EXAMPLE_ITEM, "Artículo de ejemplo");
```

Comment:
For reference, what language is this, and from what was this translation taken?

Reply:
The Spanish 'es_es' locale

Reply:
After a bit of consultation with @gigaherz and the official Crowdin translations for Spanish `es_es`, the proper and consistent translation for `Example item` is `Objeto de ejemplo`.

This means that we cannot use it as a example for how diacritics are serialized. However, since this is only an example, we needn't use an actual translation: we can use a dummy translation key, and fill in the value as something like `Example with a díacritic`.
```suggestion
    this.addItem("example.diacritic", "Example with a díacritic");
```